### PR TITLE
feat(console): path mapping migration [ part 1 ]

### PR DESCRIPTION
**Pull request by Okhelifi**
## Issue

https://graviteecommunity.atlassian.net/browse/APIM-434

## Description

The part 1 contains the new page, the edit and the delete.

https://user-images.githubusercontent.com/25704259/207895219-15d9286c-93fa-417c-ab8b-5531bd68cc04.mov

🚀 CI was able to deploy the build of this PR, so you can now try it directly [here](https://apimnightlywebui24386.z6.web.core.windows.net/apim-434-path-mapping/index.html)
_Notes_: The deployed app is linked to the management API of APIM master. (Same login and password as APIM master)

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-qfpicncxzx.chromatic.com)

### CODE REVIEW

File: gravitee-apim-console-webui/src/management/api/analytics/pathMappings/api-path-mappings.component.html [hunk 1]
```diff
@@ -16,3 +16,48 @@
...
+    <!-- Actions Column -->
+    <ng-container matColumnDef="actions">
+      <th mat-header-cell *matHeaderCellDef id="actions"></th>
+      <td mat-cell *matCellDef="let element">
+        <div class="path-mappings__table__actions">
+          <button mat-icon-button aria-label="Button to edit a path mapping" matTooltip="Edit path mapping">
+            <mat-icon svgIcon="gio:edit-pencil"></mat-icon>
+          </button>
+          <button
+            *gioPermission="{ anyOf: ['api-definition-u'] }"
```
Comment (ThibaudAV) [inline gravitee-apim-console-webui/src/management/api/analytics/pathMappings/api-path-mappings.component.html:45]: not d ?

File: gravitee-apim-console-webui/src/management/api/analytics/pathMappings/api-path-mappings-edit-dialog/api-path-mappings-edit-dialog.component.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    this.pathFormGroup = new FormGroup({
+      path: new FormControl(dialogData.path, [Validators.required, this.isUnique(dialogData.api.path_mappings)]),
+    });
+  }
+
+  public save(): void {
+    // Do something
+  }
+
+  private isUnique(paths: string[]): ValidatorFn | null {
```
Comment (ThibaudAV) [inline gravitee-apim-console-webui/src/management/api/analytics/pathMappings/api-path-mappings-edit-dialog/api-path-mappings-edit-dialog.component.ts]: Juste a wondering, about adding this to shared `utils/simpleValidator.ts` with simple `export const = (array: unknow[]) ...` something like this

File: gravitee-apim-console-webui/src/management/api/analytics/pathMappings/api-path-mappings.model.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Api } from '../../../../entities/api';
+
+export interface PathMappingDS {
+  path: string;
+}
+
+export interface ApiPathMappingsEditDialogData {
```
Comment (ThibaudAV) [inline gravitee-apim-console-webui/src/management/api/analytics/pathMappings/api-path-mappings.model.ts]: Juste a note : Personally i add this kind of type with then dialog component to have it just next to the  `@Inject(MAT_DIALOG_DATA) dialogData ...` i found it more easier to maintain

**Review by ThibaudAV — APPROVED** [review]
next is add readonly mode for k8s ?

**Comment by Okhelifi** [comment]
> next is add readonly mode for k8s ?

Yes pagination, and swagger import and then read only

**Pull request merged**